### PR TITLE
Implement Vec3 object in Lua and upgrade event callbacks

### PR DESCRIPTION
**Pull request by apachano**
Authors:
@apachano 

## Summary of changes
- Implement a vec3 class mapping to lua usertype
- Upgrade block and item events to use vec3 instead of three floats
  
## Caveats
- None

## On approval
Merge

### CODE REVIEW

File: Phoenix/Client/Source/Game.cpp [hunk 1]
```diff
@@ -99,6 +99,23 @@ Game::Game(gfx::Window* window, entt::registry* registry, bool networked)
...
+	typeVec3["y"] = &math::vec3::y;
+	typeVec3["z"] = &math::vec3::z;
+
+	m_modManager->registerFunction("voxel.map.getBlock",
+	                               [this](math::vec3 pos) {
+		                               // TODO: We should prevent this from
+		                               // being called before the map is
+		                               // initialized
+		                               auto block = m_map->getBlockAt(pos);
+		                               LOG_DEBUG("TEMP") << block->id;
```
Comment (beeperdeeper089) [inline Phoenix/Client/Source/Game.cpp:116]: Do you still need this?

File: Phoenix/Common/Include/Common/CMS/ModManager.inl [hunk 2]
```diff
@@ -96,4 +96,12 @@ namespace phx::cms
 			       "\"core.block.register\".";
 		}
 	}
-} // namespace phx::mods
+
+	template <typename T, typename... Args>
+	sol::usertype<T> ModManager::registerType(std::string name,
```
Comment (beeperdeeper089) [inline Phoenix/Common/Include/Common/CMS/ModManager.inl:101]: Const std::string&

File: Phoenix/Common/Source/Voxels/Chunk.cpp [hunk 3]
```diff
@@ -75,7 +77,8 @@ void Chunk::setBlockAt(const phx::math::vec3& position, Block newBlock)
 		}
 		if (newBlock.type->onPlace)
 		{
-			newBlock.type->onPlace(position.x, position.y, position.z);
+			LOG_DEBUG("CHUNK") << "onPlace" << newBlock.type->id;
```
Comment (beeperdeeper089) [inline Phoenix/Common/Source/Voxels/Chunk.cpp:80]: Do we need this?

**Review by beeperdeeper089 — CHANGES_REQUESTED** [review]
On phone so nothing major, I'll re-review once I'm on my computer w some time

**Comment by apachano** [comment]
I addressed those thoughts

**Review by beeperdeeper089 — APPROVED** [review]
LGTM.

**Pull request merged**